features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/Debouncer.kt

```diff
@@ -29,10 +32,12 @@ internal class Debouncer(
         }
         handler.removeCallbacksAndMessages(null)
         val timePassedSinceLastExecution = System.nanoTime() - lastTimeRecordWasPerformed
-        if (timePassedSinceLastExecution >= maxRecordDelayInNs) {
-            executeRunnable(runnable)
-        } else {
-            handler.postDelayed({ executeRunnable(runnable) }, DEBOUNCE_TIME_IN_MS)
+        runInTimeBalance {
+            if (timePassedSinceLastExecution >= maxRecordDelayInNs) {
+                executeRunnable(runnable)
+            } else {
+                handler.postDelayed({ executeRunnable(runnable) }, DEBOUNCE_TIME_IN_MS)
```

Comment:
> [!WARNING]
> Shouldn't you remove previously posted runnables, if there are multiple debounced executions ?

Reply:
this is done above at line 33

Reply:
You are right, so I changed the place where `runInTimeBalance` should wrap up. 
Now `runInTimeBalance` wrap only the function of `executeRunnable`, so we check first the condition of debounce frequency, then check if it meets the time balance.